core/lib/src/unix.rs

```diff
@@ -0,0 +1,67 @@
+use crate::error::{LaunchError, LaunchErrorKind};
+use std::path::Path;
+use std::fs::{self, File};
+
+#[doc(hidden)]
+pub fn lock_socket<P: AsRef<Path>>(path: P) -> Result<File, LaunchError> {
```

Comment:
Does this need to be `pub`? Can it be `crate` or just private?

Reply:
Can't be private because it's used in `serve_unix_socket` macro which is used in`serve` macro.

Reply:
Okay. Let's make it `crate` and remove the `doc(hidden)`.